Makefile

```diff
@@ -319,15 +309,62 @@ $(BUILT)/add-on/chrome/locale/%/loop.properties: locale/%/add-on.properties loca
 	@mkdir -p $(@D)
 	cat $^ > $@
 
+# For each of the built js file patterns, create a pattern-specific variable
+# `PREREQ` to match the prerequisite .jsx in the appropriate source directory to
+# be used during secondary expansion for the shared build recipe.
+#
+# NB: The following targets are ordered in a way that puts general rules above
+# specific rules, i.e., a target for a parent directory (which happens to match
+# all sub directories) comes before a sub directory with a different `PREREQ`.
+# This ordering can be achieved by keeping the list lexicographically sorted.
+$(BUILT)/add-on/chrome/content/modules/%.jsm: PREREQ = add-on/chrome/modules/$(@F)
+$(BUILT)/add-on/chrome/content/modules/%.js: PREREQ = add-on/chrome/modules/$(@F)
+$(BUILT)/add-on/chrome/content/panels/js/%.js: PREREQ = add-on/panels/js/$(@F)*
+$(BUILT)/add-on/chrome/content/panels/test/%.js: PREREQ = add-on/panels/test/$(@F)
+$(BUILT)/add-on/chrome/content/panels/vendor/%.js: PREREQ = add-on/panels/vendor/$(@F)*
+$(BUILT)/add-on/chrome/content/shared/js/%.js: PREREQ = shared/js/$(@F)*
+$(BUILT)/add-on/chrome/content/shared/test/%.js: PREREQ = shared/test/$(@F)
+$(BUILT)/standalone/%.js: PREREQ = standalone/$(@F)
+$(BUILT)/standalone/content/%.js: PREREQ = standalone/content/$(@F)
+$(BUILT)/standalone/content/js/%.js: PREREQ = standalone/content/js/$(@F)*
+$(BUILT)/standalone/content/shared/js/%.js: PREREQ = shared/js/$(@F)*
+$(BUILT)/standalone/content/shared/test/%.js: PREREQ = shared/test/$(@F)*
+$(BUILT)/standalone/test/%.js: PREREQ = standalone/test/$(@F)*
+$(BUILT)/ui/%.js: PREREQ = ui/$(@F)*
+$(BUILT)/ui/shared/js/%.js: PREREQ = shared/js/$(@F)*
+$(BUILT)/ui/shared/test/%.js: PREREQ = shared/test/$(@F)*
+
+# During Makefile's secondary expansion, use the prerequisite specified in the
+# pattern-specific `PREREQ` to build one `%.js` file that matches any of the
+# explicily listed targets from the `built_*_js_files` variables.
+ALL_BUILT_JS_FILES = \
+	$(built_ui_shared_js_files) \
+	$(built_add_on_shared_js_files) \
+	$(built_standalone_shared_js_files) \
+	$(built_ui_js_files) \
+	$(built_standalone_js_files) \
+	$(built_add_on_js_files) \
+	$(built_add_on_vendor_js_files)
+
+.SECONDEXPANSION:
+$(ALL_BUILT_JS_FILES): %: $$(PREREQ)
+	@mkdir -p $(@D)
+	$(BABEL) $< --out-file $@
+
+$(BUILT)/standalone/content/webappEntryPoint.js: standalone/content/webappEntryPoint.js
+	@mkdir -p $(@D)
+	$(BABEL) $< --out-file $@
```

Comment:
This seems to be already covered by ` +$(BUILT)/standalone/content/%.js: PREREQ = standalone/content/$(@F) ` ?